File: LeetCode/src/MiddleoftheLinkedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode[] result = new  ListNode[100];
        int index = 0;
        while(head!=null){
            result[index] = head;
            head = head.next;
            index ++;
        }
        
        return result[index/2];
    }
}

File: LeetCode/src/MiddleoftheLinkedList.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def middleNode(self, head: ListNode) -> ListNode:
        pointer1 = head
        pointer2 = head
        while (pointer2 != None and pointer2.next != None):
            pointer1 = pointer1.next
            pointer2 = pointer2.next.next

        return pointer1


# this is the runner technique which is the two pointer technique.